src/ClientData/DataManager.cpp

```diff
@@ -54,6 +54,12 @@ void DataManager::set_selected_thread_id(uint32_t thread_id) {
   selected_thread_id_ = thread_id;
 }
 
+void DataManager::set_selected_thread_state_slice(
+    std::optional<orbit_client_data::ThreadStateSliceInfo> selected_thread_state_slice) {
```

Comment:
I had a discussion with Henning about this. Basically it's safer to do this, in case the slice is destroyed but the pointer hasn't been set to null it would cause problems. From what I understood we should actually do it like this for the functions where we return a pointer and not a value.

Reply:
Why the slice could be destroyed? Let's briefly discuss on Monday about this @beckerhe

Reply:
Please not a pointer. We already have way too many of them. A slice is just a couple of bytes.

When we need sharing we can introduce that later.